Description: Fix -Wunused-result compiler warnings.
 In particular, print warning messages if we fail to parse note files.
Author: Doug Torrance <dtorrance@monmouthcollege.edu>
Last-Update: 2015-01-28

--- a/wmstickynotes.c
+++ b/wmstickynotes.c
@@ -412,12 +412,18 @@
 		note->id = atoi(entry->d_name);
 		if(note->id > highest_note_id) highest_note_id = note->id;
 
-		fscanf(file, "%d,%d,%d,%d,%d,%d,",
+		if (fscanf(file, "%d,%d,%d,%d,%d,%d,",
 			&(note->x), &(note->y), &(note->width), &(note->height),
-			&reserved1, &reserved2);
+			&reserved1, &reserved2) < 6) {
+			fprintf(stderr, "Failed to parse note %s: too few values", entry->d_name);
+			continue;
+		}
 
 		/* Get color name */
-		fgets(buffer, 256, file);
+		if (fgets(buffer, 256, file) == NULL) {
+			fprintf(stderr, "Failed to parse note %s: no color", entry->d_name);
+			continue;
+		}
 		/* Replace the newline with a null char */
 		buffer[strlen(buffer) - 1] = '\0';
 
